### ISSUE TITLE
SWRevealViewController Top layout guide offset 20px.

### ISSUE THREAD

**Issue by yasiriqbal** [John-Lluch/SWRevealViewController#555]
When I swipe It is all OK! But when I tap the Menu button and tap it again to close side view controller! 
The navigation bar gets 20px of upward offset! Status bar is shown but Background View(20px height) for status bar is not shown.

1) Before tapping Menu button (origional)
![1](https://cloud.githubusercontent.com/assets/11014528/13405017/5ebc5de6-df3e-11e5-9dca-8f0cd4d4e1d8.png)

2) After tapping Menu button two times, opening and closing rear view controller (without changing any view)
![2](https://cloud.githubusercontent.com/assets/11014528/13405062/8c573186-df3e-11e5-946d-05a3a40dbd5b.png)

When I rotate device then It gets adjusted again! Is there any solution or hack to avoid vertical offset?

**Comment by iDevelopper** [comment]
Is it a navigation controller?

**Comment by yasiriqbal** [comment]
Sorry for late reply! 
I am not using navigation controller!

It is a simple View Controller with a Navigation Bar! Just simple setup! When I rotate to portrait and then again rotate to landscape then subviews of front view are layout and offset is adjusted again (as i wanted)! The problem is with the TOP Layout Guide's offset! I am using Auto-layout.

**Comment by iDevelopper** [comment]
You can send me your sample project?

**Comment by yasiriqbal** [comment]
rear_view_controller has "perferStatusBarHidden -> true" and front view controller has! "preferStatusBarHidden -> false" I want to hide status bar when slide menu opens!
This works well with pan gestures on front view controller but doesn't with clicking MenuButton! 

[SWReveal Problem 2.zip](https://github.com/John-Lluch/SWRevealViewController/files/158130/SWReveal.Problem.2.zip)

Read the Comments in View Controller Files!

**Comment by yasiriqbal** [comment]
And oh yes! when you open the side bar by clicking MenuButton and "change front view (or push in case of navigation controller)" then you can see the falling/jerk of front view controller which shows the status bar!

**Comment by iDevelopper** [comment]
The status bar is normally for the entire application / screen. If you want to displace it try what i added in comment in viewWillAppear and viewWillDisappear in MenuViewController (but I don't like that).

Why do not use a UINavigationController? It works like a charm.

You can also use this library (optional) to keep the size of the navigation bar when status bar is hidden : 

https://github.com/wicharek/UINavigationBar-FixedHeightWhenStatusBarHidden i added in your project.

See in viewDidLoad method in your PollViewController.

[GoWebilc 2.zip](https://github.com/John-Lluch/SWRevealViewController/files/159035/GoWebilc.2.zip)

**Comment by yasiriqbal** [comment]
Thanks! I think you have solved the problem! I'll check it! I am using navigation controller (same problem).
I wonder why this works perfect with swipe/pan gesture and not with tapping the menu button!

**Comment by iDevelopper** [comment]
Don't understand. It works or not?

**Comment by yasiriqbal** [comment]
Yes it worked! Your solution is working and I will use this next time! but temporarily I have solve this problem with existing View Controllers by having Navigation Bar's top constraint relative to "Super View" not relative to "Top Layout Guide"!  "Top Layout Guide's" offset changes with status bar hidden or shown!

**Comment by pierrem** [comment]
I think the layout is not updated when performing a swipe gesture(adding self.revealViewController.panGestureRecognizer to my main view) , but it is updated when tapping the menu button (revealToggle). I'm using autolayout, and for me it's ok when tapping the menu button, but the view overlaps the status bar when the left drawer is opened with a swipe gesture.

**Comment by iDevelopper** [comment]
Please see #585

**Comment by pierrem** [comment]
Thank you, its working now ! And the reference to http://stackoverflow.com/a/31548575/4172423 to set the constraint to the superview help a lot !
